### ISSUE TITLE
Successful exit code if a regex is invalid

### ISSUE THREAD

**Issue by dizzersee** [awslabs/git-secrets#151]
If an invalid regex pattern is added to config, e.g. "(" not escaped, git-secrets exits with exit code 0. This is not good for integrating it into a CI/CD-pipeline. As there is an error, shouldn't it exit with another exit code?

**Comment by sparr** [comment]
I agree that git-secrets should exit with a non-zero exit code in the case of encountering an error in a pattern.

Also, --add should attempt to validate the pattern, although invalid patterns could still be added via `git config` or directly editing the config file.
